src/app/components/GameSetup.tsx

```diff
@@ -1,57 +1,80 @@
-import React from "react";
+import React, { useState } from "react";
 import { Game } from "../page";
-import { FormEvent } from "react";
 
 type Props = {
-  gameSetup: Game;
-  setGameSetup: React.Dispatch<React.SetStateAction<Game>>;
+  gameSetup: Game | undefined;
+  setGameSetup: React.Dispatch<React.SetStateAction<Game | undefined>>;
 };
 
-function GameSetup({ setGameSetup, gameSetup }: Props) {
-  function handleSubmit(e: FormEvent<HTMLFormElement>) {
-    e.preventDefault();
-// sets state that carries the game information (teams, duration, etc) 
-    setGameSetup({
-      periods: e.target.periods.value,
-      length: e.target.duration.value,
-      teamA: e.target.teamA.value,
-      teamB: e.target.teamB.value,
-    });
-    console.log(gameSetup);
+function GameSetup({ setGameSetup }: Props) {
+  const [formData, setFormData] = useState({
+    periods: 2,
+    duration: 10,
+    teamA: "",
+    teamB: "",
+  });
+
+  function handleChange(
+    event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) {
+    const { name, value } = event.target;
+    setFormData((prev) => ({ ...prev, [name]: value }));
```

Comment:
This is a bit tricky if you haven't done it before but yea essentially updating the state object immutably and also generalising the name and value for any potential form element